src/jit/ee_il_dll.cpp

```diff
@@ -146,6 +146,47 @@ ICorJitCompiler* __stdcall getJit()
     return(ILJitter);
 }
 
+static void ensureInitializationOnFirstCompilation(ICorJitInfo* compHnd)
+{
+    static Volatile<bool> s_isInitialized;
+    static CritSecObject s_initializationLock;
+
+    if (s_isInitialized.Load())
+    {
+        _ASSERTE(JitConfig.isInitialized());
+        return;
+    }
+
+    ClrEnterCriticalSection(s_initializationLock.Val());
+
+    if (!s_isInitialized.LoadWithoutBarrier())
+    {
+        _ASSERTE(!JitConfig.isInitialized());
+
+        // NOTE: this code currently assumes that the ICorStaticInfo::get*ConfigValue
+        //       is idempotent for the same arguments not only across calls, but also
+        //       across all ICorJitInfo instances that the JIT may be exposed to
+        //       during its lifetime. Removing this assumption is possible, but would
+        //       have wide-reaching effects: config values could no longer be stored
+        //       in a simple cache, and any code that uses config values that doesn't
+        //       have access to an ICorJitInfo instance would need to be refactored.
+        JitConfig.initialize(compHnd, ProcessHeapAllocator::Singleton());
```

Comment:
It would be great to address this - VSO 101430 is still open, and assigned to me. We get asm diffs between ret and chk because we wind up with config values that are persisted across captured contexts from different processes.